### ISSUE TITLE
Unable to create working Elsa Server application as per official documentation

### ISSUE THREAD

**Issue by francotiveron** [elsa-workflows/elsa-core#4731]
I did exactly, step by step as described in https://v3.elsaworkflows.io/docs/installation/elsa-server. I encountered the following issues:

1) Build Error

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	NU1107	Version conflict detected for Microsoft.Extensions.DependencyInjection.Abstractions. Install/reference Microsoft.Extensions.DependencyInjection.Abstractions 8.0.0 directly to project HorthFinancial.NinjaTrader.Workflows.Server to resolve this issue. 
 HorthFinancial.NinjaTrader.Workflows.Server -> Elsa 3.0.0-preview.945 -> Microsoft.Extensions.DependencyInjection.Abstractions (>= 8.0.0) 
 HorthFinancial.NinjaTrader.Workflows.Server -> Elsa.Identity 3.0.0-preview-preview-3.0.0-740-hotfix-1.771 -> Elsa.Api.Common 3.0.0-preview-preview-3.0.0-740-hotfix-1.771 -> FastEndpoints.Swagger 5.13.0 -> NSwag.AspNetCore 13.19.0 -> NSwag.Generation.AspNetCore 13.19.0 -> Microsoft.Extensions.DependencyInjection.Abstractions (>= 7.0.0 && < 8.0.0).	HorthFinancial.NinjaTrader.Workflows.Server	C:\Root\Work\Horth\Work\Ninja\Code\HorthFinancial.NinjaTrader.Workflows.Server\HorthFinancial.NinjaTrader.Workflows.Server.csproj	1	
```

As required I added the reference

```
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <Nullable>enable</Nullable>
    <ImplicitUsings>enable</ImplicitUsings>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Elsa" Version="3.0.0-preview.945" />
    <PackageReference Include="Elsa.CSharp" Version="3.0.0-preview.945" />
    <PackageReference Include="Elsa.EntityFrameworkCore" Version="3.0.0-preview-preview-3.0.0-740-hotfix-1.771" />
    <PackageReference Include="Elsa.EntityFrameworkCore.Sqlite" Version="3.0.0-preview-preview-3.0.0-740-hotfix-1.771" />
    <PackageReference Include="Elsa.Identity" Version="3.0.0-preview-preview-3.0.0-740-hotfix-1.771" />
    <PackageReference Include="Elsa.Scheduling" Version="3.0.0-preview-preview-3.0.0-740-hotfix-1.771" />
	<PackageReference Include="Elsa.Workflows.Api" Version="3.0.0-preview-preview-3.0.0-740-hotfix-1.771" />
	<PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="8.0.0" />
  </ItemGroup>

</Project>
```

2) Build fails again
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	NU1107	Version conflict detected for Microsoft.Extensions.Options. Install/reference Microsoft.Extensions.Options 8.0.0 directly to project HorthFinancial.NinjaTrader.Workflows.Server to resolve this issue. 
 HorthFinancial.NinjaTrader.Workflows.Server -> Elsa 3.0.0-preview.945 -> Microsoft.Extensions.Hosting.Abstractions 8.0.0 -> Microsoft.Extensions.Diagnostics.Abstractions 8.0.0 -> Microsoft.Extensions.Options (>= 8.0.0) 
 HorthFinancial.NinjaTrader.Workflows.Server -> Elsa.Identity 3.0.0-preview-preview-3.0.0-740-hotfix-1.771 -> Elsa.Api.Common 3.0.0-preview-preview-3.0.0-740-hotfix-1.771 -> FastEndpoints.Swagger 5.13.0 -> NSwag.AspNetCore 13.19.0 -> NSwag.Generation.AspNetCore 13.19.0 -> Microsoft.Extensions.Options (>= 7.0.0 && < 8.0.0).	HorthFinancial.NinjaTrader.Workflows.Server	C:\Root\Work\Horth\Work\Ninja\Code\HorthFinancial.NinjaTrader.Workflows.Server\HorthFinancial.NinjaTrader.Workflows.Server.csproj	1	
```

I added again the required reference

```
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <Nullable>enable</Nullable>
    <ImplicitUsings>enable</ImplicitUsings>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Elsa" Version="3.0.0-preview.945" />
    <PackageReference Include="Elsa.CSharp" Version="3.0.0-preview.945" />
    <PackageReference Include="Elsa.EntityFrameworkCore" Version="3.0.0-preview-preview-3.0.0-740-hotfix-1.771" />
    <PackageReference Include="Elsa.EntityFrameworkCore.Sqlite" Version="3.0.0-preview-preview-3.0.0-740-hotfix-1.771" />
    <PackageReference Include="Elsa.Identity" Version="3.0.0-preview-preview-3.0.0-740-hotfix-1.771" />
    <PackageReference Include="Elsa.Scheduling" Version="3.0.0-preview-preview-3.0.0-740-hotfix-1.771" />
	<PackageReference Include="Elsa.Workflows.Api" Version="3.0.0-preview-preview-3.0.0-740-hotfix-1.771" />
	<PackageReference Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="8.0.0" />
    <PackageReference Include="Microsoft.Extensions.Options" Version="8.0.0" />
  </ItemGroup>

</Project>
```

3) Build succeeds even though there are warnings

```
Detected package version outside of dependency constraint: NSwag.Generation.AspNetCore 13.19.0 requires Microsoft.Extensions.DependencyInjection.Abstractions (>= 7.0.0 && < 8.0.0) but version Microsoft.Extensions.DependencyInjection.Abstractions 8.0.0 was resolved.

Detected package version outside of dependency constraint: NSwag.Generation.AspNetCore 13.19.0 requires Microsoft.Extensions.Options (>= 7.0.0 && < 8.0.0) but version Microsoft.Extensions.Options 8.0.0 was resolved.
```

4) Application fails to start, triggering exception

```
System.TypeLoadException
  HResult=0x80131522
  Message=Could not load type 'Elsa.Workflows.Features.WorkflowsFeature' from assembly 'Elsa.Workflows.Core, Version=3.0.0.0, Culture=neutral, PublicKeyToken=null'.
  Source=System.Private.CoreLib
  StackTrace:
   at System.Reflection.RuntimeAssembly.GetTypeCore(String typeName, ReadOnlySpan`1 nestedTypeNames, Boolean throwOnError, Boolean ignoreCase)
   at System.Reflection.TypeNameParser.GetType(String typeName, ReadOnlySpan`1 nestedTypeNames, String assemblyNameIfAny)
   at System.Reflection.TypeNameParser.Parse()
   at System.Reflection.TypeNameParser.GetTypeHelper(Char* pTypeName, RuntimeAssembly requestingAssembly, Boolean throwOnError, Boolean requireAssemblyQualifiedName)
   at System.Reflection.CustomAttribute._CreateCaObject(RuntimeModule pModule, RuntimeType type, IRuntimeMethodInfo pCtor, Byte** ppBlob, Byte* pEndBlob, Int32* pcNamedArgs)
   at System.Reflection.CustomAttribute.AddCustomAttributes(ListBuilder`1& attributes, RuntimeModule decoratedModule, Int32 decoratedMetadataToken, RuntimeType attributeFilterType, Boolean mustBeInheritable, ListBuilder`1 derivedAttributes)
   at System.Reflection.CustomAttribute.GetCustomAttributes(RuntimeType type, RuntimeType caType, Boolean inherit)
   at System.Attribute.GetCustomAttributes(MemberInfo element, Type attributeType, Boolean inherit)
   at System.Reflection.CustomAttributeExtensions.GetCustomAttributes[T](MemberInfo element)
   at Elsa.Features.Implementations.Module.GetDependencyTypes(Type type)
   at System.Linq.Enumerable.SelectManySingleSelectorIterator`2.MoveNext()
   at System.Linq.Enumerable.ConcatIterator`1.MoveNext()
   at System.Linq.Enumerable.SelectManySingleSelectorIterator`2.MoveNext()
   at System.Linq.Enumerable.ConcatIterator`1.MoveNext()
   at System.Collections.Generic.HashSet`1.UnionWith(IEnumerable`1 other)
   at System.Collections.Generic.HashSet`1..ctor(IEnumerable`1 collection, IEqualityComparer`1 comparer)
   at System.Linq.Enumerable.ToHashSet[TSource](IEnumerable`1 source, IEqualityComparer`1 comparer)
   at Elsa.Features.Implementations.Module.GetFeatureTypes()
   at Elsa.Features.Implementations.Module.Apply()
   at Elsa.Extensions.ModuleExtensions.AddElsa(IServiceCollection services, Action`1 configure)
   at Program.<Main>$(String[] args) in C:\Root\Work\Horth\Work\Ninja\Code\HorthFinancial.NinjaTrader.Workflows.Server\Program.cs:line 7
```

**Comment by sfmskywalker** [comment]
Thanks for reporting @francotiveron. The documentation has been updated. Please give it a try and reopen this issue if you identify more issues with the walkthrough.

**Comment by sfmskywalker** [comment]
PS. Make sure to reference the same versions of all Elsa packages. Mixed versions will cause issues. For example, this won't work:

```xml
<PackageReference Include="Elsa.CSharp" Version="3.0.0-preview.945" />
<PackageReference Include="Elsa.EntityFrameworkCore" Version="3.0.0-preview-preview-3.0.0-740-hotfix-1.771" />
```

But this works:

```xml
<PackageReference Include="Elsa.CSharp" Version="3.0.0-preview.945" />
<PackageReference Include="Elsa.EntityFrameworkCore" Version="3.0.0-preview.945" />
```

And now that 3.0 is released to NuGet, you can reference 3.0.0:

```xml
<PackageReference Include="Elsa.CSharp" Version="3.0.0" />
<PackageReference Include="Elsa.EntityFrameworkCore" Version="3.0.0" />
```
